src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectWriter.cs

```diff
@@ -984,7 +984,7 @@ private long GetAssemblyId(WriteObjectInfo objectInfo)
             {
                 assemId = 0;
             }
-            else if (assemblyString.Equals(Converter.s_urtAssemblyString))
+            else if (assemblyString.Equals(Converter.s_urtAssemblyString) || assemblyString.Equals(Converter.s_urtAlternativeAssemblyString))
```

Comment:
Either my understanding of TypeForwardedFrom is incomplete and it isn't used at the time of serialization, or the code which has the assembly stripping optimization doesn't consider TypeForwardedFrom, as I mentioned in an edit to the above comment.